### ISSUE TITLE
Default methods in javascript wrapper interfaces are not supported

### ISSUE THREAD

**Issue by bennylut** [konsoletyper/teavm#96]
When defining an javascript wrapper interface one cannot use default methods. 
 e.g., The program:

```
public Interface IfcWithDefaultMethod extends JSObject {
  default void printHello(){
    System.out.println("Hello");
  }
}

public class Main {
  public static void main(String[] args){
    IfcWithDefaultMethod i = ... // get an instance from the global environment
    i.printHello();
  }
}
```

is resulting with an exception at run time: 
"bwt_Main97.window99.test.hello is not a function"

where "test" is the name of the wrapped instance extracted from the global environment.
"test" did not contain the function printHello which therefore should have been resulted in the invocation of the default method.

**Comment by konsoletyper** [comment]
The stable version of TeaVM does not support Java 8 default interfaces for wrapping JavaScript. The recent version of TeaVM does, but there's no documentation yet (as it is not stable), and several restriction. One restriction (one that you violate here) is that default methods (as well as non-abstract methods of abstract classes) must be final and must not override methods of parent classes/interfaces. Sorry, but it's not possible to remove this restriction. I'll update documentation and improve TeaVM to report about violations before releasing 0.3.0

**Comment by bennylut** [comment]
I see, 
for abstract classes it make sense, but how can you define final default method in an interface?

**Comment by konsoletyper** [comment]
You are right, there is no way. So you should either refuse from interfaces or from default methods.

**Comment by konsoletyper** [comment]
Here is my proposal. I can invent my own `@Final` annotation and check it with TeaVM compiler. So you could just add this annotation to your `default` methods.

**Comment by bennylut** [comment]
Yes, I actually just working my way up TeaVM JavaScriptNativeProcessor code, and trying to add a support for a `@Default` annotation that I hope to translate to javascript by doing something along the following pseudo code: 
Java: `obj.method(params)`
Javascript: `$obj.$method? $obj.$method($params): $static_$method($obj, $params)`

I still trying to grasp the concepts there (some documentation might help ;) ) , so it may take sometime..

**Comment by konsoletyper** [comment]
You should not do this way. I'll write some thoughs, may be privately. Of course, I can implement this feature myself.

**Comment by bennylut** [comment]
OK, Not a problem.
